File: ios/Zenith/BodyMap3DViewManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(BodyMap3DViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(snapshotJson, NSString)
RCT_EXPORT_VIEW_PROPERTY(stimulusLensJson, NSString)
RCT_EXPORT_VIEW_PROPERTY(regionPanelsJson, NSString)
RCT_EXPORT_VIEW_PROPERTY(activeLens, NSString)
RCT_EXPORT_VIEW_PROPERTY(overlayMode, NSString)
RCT_EXPORT_VIEW_PROPERTY(cameraPreset, NSString)
RCT_EXPORT_VIEW_PROPERTY(selectedRegionId, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(onRegionPress, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onInteractionStateChange, RCTBubblingEventBlock)

@end

File: ios/Zenith/BodyMap3DViewManager.swift
import Foundation
import React

@objc(BodyMap3DViewManager)
class BodyMap3DViewManager: RCTViewManager {
  override static func requiresMainQueueSetup() -> Bool {
    true
  }

  override func view() -> UIView! {
    BodyMap3DView()
  }
}
